client/app/components/Paginator/index.js

```diff
@@ -8,11 +9,12 @@ import styles from './styles.scss';
 // Components
 import RangeButtons from './RangeButtons';
 
-const Paginator = ({ currPage, numPages, getPosts }) => {
+const Paginator = ({ currPage, numPages }) => {
   // Wrapper function to prevent out of range page requests
-  const getPostsSafe = (target) => {
-    if (target > 0 && target <= numPages) {
-      getPosts(target);
+  const getPostsSafe = (e, target) => {
+    if (!(target > 0 && target <= numPages)) {
+      console.warn(`User attempted to request invalid page number: ${target}`);
+      e.preventDefault();
     }
```

Comment:
you should move this if outside of the function. You shouldn't call an empty function